Suppress `Error loading HRE` from telemetry

We handle and display loading errors in the hardhat config, but send the exception on to sentry our telemetry and exception reporting system. We should suppress it from going to sentry, so our telemetry is not full of variants of:

> Error loading HRE: Error: Cannot find module '@nomicfoundation/hardhat-toolbox'

--- a/server/src/telemetry/SentryServerTelemetry.ts
+++ b/server/src/telemetry/SentryServerTelemetry.ts
@@ -8,6 +8,8 @@ import { ServerState } from "../types";
 import { Analytics } from "../analytics/types";
 import { Telemetry, TrackingResult } from "./types";
 
+import { sentryEventFilter } from "./sentryEventFilter";
+
 const SENTRY_CLOSE_TIMEOUT = 2000;
 
 export class SentryServerTelemetry implements Telemetry {
@@ -18,7 +20,12 @@ export class SentryServerTelemetry implements Telemetry {
   private heartbeatInterval: NodeJS.Timeout | null;
   private heartbeatPeriod: number;
 
-  constructor(dsn: string, heartbeatPeriod: number, analytics: Analytics) {
+  constructor(
+    dsn: string,
+    heartbeatPeriod: number,
+    analytics: Analytics,
+    public eventFilter = sentryEventFilter
+  ) {
     this.dsn = dsn;
     this.heartbeatPeriod = heartbeatPeriod;
     this.serverState = null;
@@ -57,7 +64,10 @@ export class SentryServerTelemetry implements Telemetry {
           client: clientName,
         },
       },
-      beforeSend: (event) => (isTelemetryEnabled(serverState) ? event : null),
+      beforeSend: (event) =>
+        isTelemetryEnabled(serverState) && this.eventFilter(event)
+          ? event
+          : null,
     });
 
     this.analytics.init(machineId, extensionVersion, serverState, clientName);

--- a/server/src/telemetry/sentryEventFilter.ts
+++ b/server/src/telemetry/sentryEventFilter.ts
@@ -1,0 +1,16 @@
+import * as Sentry from "@sentry/node";
+
+const EXCLUDE_PATTERNS = [/error loading hre/gi];
+
+export function sentryEventFilter(event: Sentry.Event): boolean {
+  for (const value of event.exception?.values ?? []) {
+    if (
+      value.type === "Error" &&
+      EXCLUDE_PATTERNS.some((pattern) => value.value?.match(pattern))
+    ) {
+      return false;
+    }
+  }
+
+  return true;
+}

